src/tasks/JsonToItemsTaskFactory/JsonToItemsTaskFactory.cs

```diff
@@ -0,0 +1,379 @@
+                    if (jsonModel?.Properties != null &&  jsonModel.Properties.TryGetValue(property.Name, out var value))
+                    {
+                        return value;
+                    }
+                    Log.LogError("Property {0} not found in {1}", property.Name, jsonFilePath);
+                    throw new Exception();
+                }
+                else
+                {
+                    if (jsonModel?.Items != null && jsonModel.Items.TryGetValue(property.Name, out var itemModels))
+                    {
+                        return ConvertItems (itemModels);
+                    }
+
+                }
+                return null;
+            }
+
+            public static ITaskItem[] ConvertItems(JsonModelItem[] itemModels)
+            {
+                var items = new ITaskItem[itemModels.Length];
+                for (int i = 0; i < itemModels.Length; i++)
+                {
+                    var itemModel = itemModels[i];
+                    var item = new TaskItem(itemModel.Identity);
+                    if (itemModel.Metadata != null)
+                    {
+                        foreach (var metadata in itemModel.Metadata)
+                        {
+                            if (string.Equals(metadata.Key, "Identity", StringComparison.OrdinalIgnoreCase))
+                                continue;
+                            item.SetMetadata(metadata.Key, metadata.Value);
+                        }
+                    }
+                    items[i] = item;
+                }
+                return items;
+            }
+
+            public void SetPropertyValue (TaskPropertyInfo property, object? value)
+            {
+                if (logDebugTask) Log.LogMessage("SetPropertyValue called with {0}", property.Name);
+                if (property.Name == "JsonFilePath")
+                {
+                    jsonFilePath = (string)value!;
+                }
+                else
+                    throw new Exception ($"JsonToItemsTask {TaskName} cannot set property {property.Name}");
+            }
+
+        }
+
+        public class JsonModelRoot
+        {
+            [JsonConverter(typeof(CaseInsensitiveDictionaryConverter))]
+            public Dictionary<string, string>?  Properties {get; set;}
+            public Dictionary<string, JsonModelItem[]>? Items {get; set;}
+
+            public JsonModelRoot() {}
+        }
+
+        [JsonConverter(typeof(JsonModelItemConverter))]
+        public class JsonModelItem
+        {
+            public string? Identity {get; set;}
+            // n.b. will  be deserialized case insensitive
+            public Dictionary<string, string>? Metadata {get;  set; }
+        }
+
+        public class CaseInsensitiveDictionaryConverter : JsonConverter<Dictionary<string, string>>
+        {
+            public override Dictionary<string, string> Read (ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
```

Comment:
Is this the method being overridden here? https://docs.microsoft.com/en-us/dotnet/api/system.text.json.serialization.jsonconverter-1.read?view=net-5.0#system-text-json-serialization-jsonconverter-1-read(system-text-json-utf8jsonreader-system-type-syst

Should the return type be nullable then?

Reply:
Apparently not. If I change the return type to `Dictionary<string,string>?`, I get (probably from the net472 build):

```
/Users/alklig/work/dotnet-runtime/runtime-bugs/src/tasks/JsonToItemsTaskFactory/JsonToItemsTaskFactory.cs(330,57): error CS8764: Nullability of return type doesn't match overridden member (possibly because of nullability attributes). [/Users/alklig/work/dotnet-runtime/runtime-bugs/src/tasks/JsonToItemsTaskFactory/JsonToItemsTaskFactory.csproj]
```